### DISCUSSION TITLE
Using Breeze / Filament / Livewire  together. ( comments need )

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#9007] [Help]
I use Breeze 'Blade with Alpine' for client layout / authentication and Filament for backend. 

And for convenient data handling, I use Livewire. 

It is easy to install and setup Breeze / Livewire together .

But when Filament are installed and setup together, it changes a lot of settings of Breeze.

Maybe I can say that Filament is not compatible with Breeze.

So I wrote list to fix to make it work together.

Actually I am not 100% sure that it is working well. !! 

I hope it will be helpful to someone and to me also.

## Installation

1. Install Laravel Breeze / Laravel Filament / Livewire 

    ```bash
    composer require laravel/breeze --dev
    composer require laravel/filament
    composer require livewire/livewire

    ```
2. Setup Laravel Filament

    ```bash
    composer require laravel/filament
    php artisan filament:install --panels
    ```
3. Setup Laravel Breeze

    ```bash
    php artisan breeze:install

    // choose 'Blade with Alpine' option 
    // --> Error 

    ```
4. You would see error. Because Filament has changed settings for Breeze installation. We fix it.

    ```bash
    vi resources/js/app.js
    ```

    ```js
    // app.js
    import './bootstrap';

    // Because I installed livewire/livewire V3 on root, Alpinejs  is loaded from livewire 3 .

    // import Alpine from 'alpinejs';
    // window.Alpine = Alpine;
    // Alpine.start();
    ```

    ```js
    // bootstrap.js

    import axios from 'axios';
    window.axios = axios;
    window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
    
    // Echo.js part is removed.
    ```

5. install axios which is removed by 'php artisan filament:install'.
    ```bash
        npm install axios
    ```

6. setup tailwind.config.js

```js
// tailwind.config.js for using breeze default theme

import defaultTheme from 'tailwindcss/defaultTheme';
import forms from '@tailwindcss/forms';

/** @type {import('tailwindcss').Config} */
export default {
    content: [
        './vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php',
        './storage/framework/views/*.php',
        './resources/views/**/*.blade.php',
        './app/Filament/**/*.php',
        './/views/**/*.blade.php',
        './vendor/filament/**/*.blade.php',

    ],

    theme: {
        extend: {
            fontFamily: {
                sans: ['Figtree', ...defaultTheme.fontFamily.sans],
            },
        },
    },

    plugins: [forms],
};

// If you want to use filament's tailwind.config.js, you can use it with this.

// import preset from './vendor/filament/support/tailwind.config.preset'

// export default {
//     presets: [preset],
//          ;
//          ;
// }
```

7. setup vite.config.js

```js
// vite.config.js
import { defineConfig } from 'vite';
import laravel, { refreshPaths } from 'laravel-vite-plugin'

export default defineConfig({
    plugins: [
        laravel({
            input: [
                'resources/css/app.css',
                'resources/js/app.js',
            ],
            refresh: [
                ...refreshPaths,
                'app/Filament/**',
                'app/Forms/Components/**',
                'app/Livewire/**',
                'app/Infolists/Components/**',
                'app/Providers/Filament/**',
                'app/Tables/Columns/**',
            ],
        }),
    ],
});
```

8. test 
```bash 
    php artisan serve
    
    # ( on another terminal)
    npm run build 
```

9. layout settings ( additional )
    - default layouts of each packages are followings .
        breeze  - resources/views/layouts/app.blade.php
        livewire - resources/views/components/layouts/app.blade.php
        filament - resources/views/vendor/filament-panels/components/layout.blade.php
        ( you need 'php artisan vendor:publish --tag=filament-panels-views' to see it.)
    
        - You can set Livewire use breeze layout.
    
    ```php
    // livewire.php 
    // 'layout' => 'components.layouts.app',
    'layout' => 'layouts.app',
    ```

If you find something wrong or missing, please leave a comment.

As I said above, I am not 100% sure that it is working well.

Thanks for reading and comments to improve it.

**Comment by krekas** [comment]
You clearly are doing something wrong because filament shouldn't change anything. It only adds panel provider and publishes assets. Nothing more. And I don't understand why you install axios? It's not needed. The only thing needs to be done and it's because you use Livewire v3 is to remove Alpine.js that comes from breeze.
As for layouts. DON'T PUBLISH FILAMENT VIEWS. There's a really big chance after update something won't work.

**Comment by snarexx** [comment]
Same here. I got the same problem. How did you resolve this? @magic-thomas

**Reply by krekas** [reply]
What problems? Filament doesn't modify anything within breeze, and breeze doesn't within filament.

**Reply by snarexx** [reply]
@krekas It changes some config.js files, as  @magic-thomas explained, so that it breaks the breeze login page CSS styles. There may be a conflict between styles somehow. I will share a repo that reproduces this problem for you to check.

**Reply by krekas** [reply]
No, it doesn't change. You do something wrong

**Reply by krekas** [reply]
So I just created a new laravel project with blade breeze. Then installed filament. All works and no style issues.

**Reply by ossipov** [reply]
> So I just created a new laravel project with blade breeze. Then installed filament. All works and no style issues.

Try installing fillament first, then breeze after. 
Breeze changes `/resources/js/app.js` and adds:
```js
import Alpine from 'alpinejs';
window.Alpine = Alpine;
Alpine.start();
```

It breaks styling of fillament panels. Also if you comment it out Alpine wouldn't work on breeze side.

To fix it, you can change path to dedicated js for Fillament at Panel Service Provider.
```js
    public function register(): void
    {
        parent::register();
        FilamentView::registerRenderHook('panels::body.end', fn(): string => Blade::render("@vite('resources/js/app-fillament.js')"));
    }
```

**Reply by liamseys** [reply]
> @krekas It changes some config.js files, as @magic-thomas explained, so that it breaks the breeze login page CSS styles. There may be a conflict between styles somehow. I will share a repo that reproduces this problem for you to check.

Did you manage to fix this? I think I installed Filament first and then installed Breeze. When using the Breeze login pages for the first time to sign into a panel, the JavaScript gets messed up, and the user interface looks glitchy. I can't seem to find the problem.

**Reply by krekas** [reply]
> > @krekas It changes some config.js files, as @magic-thomas explained, so that it breaks the breeze login page CSS styles. There may be a conflict between styles somehow. I will share a repo that reproduces this problem for you to check.
> 
> Did you manage to fix this? I think I installed Filament first and then installed Breeze. When using the Breeze login pages for the first time to sign into a panel, the JavaScript gets messed up, and the user interface looks glitchy. I can't seem to find the problem.

When using breeze remove alpine first

**Reply by liamseys** [reply]
@krekas My app.js only contains ```import './bootstrap';```, I think AlpineJS is directly loaded from Livewire.

**Reply by krekas** [reply]
> @krekas My app.js only contains `import './bootstrap';`, I think AlpineJS is directly loaded from Livewire.

then you are doing something wrong. what wrong, it's hard to say. As everything works 100%. They have separate style files. Unless you are using forms/tables/notification in your app then do everything from the docs.

**Comment by janyksteenbeek** [comment]
Filament and Breeze are conflicting because of Livewire's navigating with wire:navigate. When it loads Filament through wire navigate under the hood, it misses critical JS files that are not loaded because of the way wire:navigate works.

In your authentication views (`login.blade.php`, `register.blade.php` etc), get rid of the `navigate: true` var of the `redirectIntended` function. This will 'hard' refresh the page after you log in and load all assets correctly.

For example, in `login.blade.php`
```php
 public function login(): void
    {
        $this->validate();

        $this->form->authenticate();

        Session::regenerate();

        $this->redirectIntended(default: route('dashboard', absolute: false), navigate: true);
    }
```

will become
```php
 public function login(): void
    {
        $this->validate();

        $this->form->authenticate();

        Session::regenerate();

        $this->redirectIntended(default: route('dashboard', absolute: false));
    }
````
